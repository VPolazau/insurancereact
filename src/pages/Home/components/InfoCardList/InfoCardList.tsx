import React from 'react';
import cn from 'classnames';

import { InfoCard } from '../InfoCard';
import { Typography } from '../../../../components';
import { CARDS } from '../../../../utils/constants';
import { decodeHtmlEntities } from '../../../../utils/utilities';

import styles from './styles.module.css';

export interface IInfoCardList {
    className?: string;
}

export const InfoCardList: React.FC<IInfoCardList> = ({ className }) => {
    return (
        <div className={cn(styles.infoCardListBasic, className)} data-testid="infoCardList">
            <Typography variant="h2">Наши преимущества</Typography>
            <div className={styles.cardListWrapper}>
                {CARDS.map((card, index) => (
                    <InfoCard key={index}>
                        <>
                            <Typography variant="h3">{card.title}</Typography>
                            <Typography variant="medium">{decodeHtmlEntities(card.description)}</Typography>
                        </>
                    </InfoCard>
                ))}
            </div>
        </div>
    );
};
