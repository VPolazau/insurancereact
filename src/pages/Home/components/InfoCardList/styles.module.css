.infoCardListBasic {
    display: flex;
    flex-direction: column;
    margin: 0 auto;
    justify-content: center;
    width: 100%;
    padding: 113px 24px;
    /* ширина по макету + 2 падинга */
    max-width: calc(1218px + 48px);
}

.cardListWrapper {
    display: flex;
    flex-direction: row;
    justify-content: center;
    flex-wrap: wrap;
    margin-top: 24px;
    width: 100%;
    gap: 30px;    
}

@media screen and (max-width: 1270px) {
    .infoCardListBasic {
        padding: 65px 24px 80px;
    } 

   .cardListWrapper {
        gap: 24px
    }
}

@media screen and (max-width: 480px) {
    .infoCardListBasic {
        padding: 40px 16px 56px;
    }
    .cardListWrapper{
        flex-direction: column;
    }
}