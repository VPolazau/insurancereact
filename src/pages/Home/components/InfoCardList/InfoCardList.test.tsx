import { render, screen } from '@testing-library/react';

import { InfoCardList } from './InfoCardList';

describe('InfoCardList', () => {
    it('InfoCardList with mock className', () => {
        render(<InfoCardList className="mockClassInfoCardList" />);
        expect(screen.getByTestId('infoCardList').classList.contains('mockClassInfoCardList')).toBe(true);
        expect(screen.getByTestId('infoCardList').classList.contains('infoCardListBasic')).toBe(true);
        expect(screen.getByTestId('infoCardList')).toBeInTheDocument();
        expect(screen.getByTestId('infoCardList')).toMatchSnapshot();
    });
});
