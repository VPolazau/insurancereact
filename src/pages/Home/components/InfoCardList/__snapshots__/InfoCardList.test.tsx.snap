// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoCardList InfoCardList with mock className 1`] = `
<div
  class="infoCardListBasic mockClassInfoCardList"
  data-testid="infoCardList"
>
  <div
    class="text h2"
    data-testid="typographyComponent"
  >
    Наши преимущества
  </div>
  <div
    class="cardListWrapper"
  >
    <div
      class="infoCardBasic"
      data-testid="infoCard"
    >
      <div
        class="rectangle"
      />
      <div
        class="text h3"
        data-testid="typographyComponent"
      >
        Широкий спектр услуг
      </div>
      <div
        class="text medium"
        data-testid="typographyComponent"
      >
        Мы заботимся о Вашей безопасности
      </div>
    </div>
    <div
      class="infoCardBasic"
      data-testid="infoCard"
    >
      <div
        class="rectangle"
      />
      <div
        class="text h3"
        data-testid="typographyComponent"
      >
        Чат поддержки
      </div>
      <div
        class="text medium"
        data-testid="typographyComponent"
      >
        Получайте быстро и удобно ответы на вопросы
      </div>
    </div>
    <div
      class="infoCardBasic"
      data-testid="infoCard"
    >
      <div
        class="rectangle"
      />
      <div
        class="text h3"
        data-testid="typographyComponent"
      >
        Личный кабинет
      </div>
      <div
        class="text medium"
        data-testid="typographyComponent"
      >
        Управляйте своей страховкой легко и удобно
      </div>
    </div>
    <div
      class="infoCardBasic"
      data-testid="infoCard"
    >
      <div
        class="rectangle"
      />
      <div
        class="text h3"
        data-testid="typographyComponent"
      >
        Выгодное предложение
      </div>
      <div
        class="text medium"
        data-testid="typographyComponent"
      >
        Страховая сумма больше, а взнос меньше
      </div>
    </div>
  </div>
</div>
`;
